### ISSUE TITLE
Library not loaded: @rpath/AppThinning.framework/Versions/A/AppThinning (Xcode 16)

### ISSUE THREAD

**Issue by lszl84** [dotnet/maui#24799]
### Description

Xcode 16 is now officially released  and **required for AppStore distribution** 

<img width="996" alt="Screenshot 2024-09-17 at 10 48 01" src="https://github.com/user-attachments/assets/1bb79a92-1fdb-4b51-bbc8-b30077301023">

However, after updating to Xcode 16, the MAUI apps stop building (VSCode, MacOS). I've read somewhere that Xcode support is planned for .NET 9, to be released in November. Does that mean we won't be able to publish MAUI apps for next few months?

```
-----------------------------------------------------------------------------
You may only use the Microsoft Visual Studio .NET/C/C++ Debugger (vsdbg) with
Visual Studio Code, Visual Studio or Visual Studio for Mac software to help you
develop and test your applications.
------------------------------------------------------------------------------
dlopen(/Applications/Xcode.app/Contents/SharedFrameworks/IDEDistribution.framework/Versions/A/IDEDistribution, 0x0001): Library not loaded: @rpath/AppThinning.framework/Versions/A/AppThinning

  Referenced from: <33FF2F3B-A96F-37B4-BA4E-887BD882BF9D> /Applications/Xcode.app/Contents/SharedFrameworks/IDEDistribution.framework/Versions/A/IDEDistribution

  Reason: tried: '/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8030/tools/lib/mlaunch/mlaunch.app/Contents/Frameworks/AppThinning.framework/Versions/A/AppThinning' (no such file), '/Applications/Xcode.app/Contents/SharedFrameworks/IDEDistribution.framework/Versions/A/Frameworks/AppThinning.framework/Versions/A/AppThinning' (no such file), '/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8030/tools/lib/mlaunch/mlaunch.app/Contents/Frameworks/AppThinning.framework/Versions/A/AppThinning' (no such file), '/Applications/Xcode.app/Contents/SharedFrameworks/IDEDistribution.framework/Versions/A/Frameworks/AppThinning.framework/Versions/A/AppThinning' (no such file), '/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net8.0_17.5/17.5.8030/tools/lib/mlaunch/mlaunch.app/Contents/MonoBundle/AppThinning.framework/Versions/A/AppThinning' (no such file)

        

error HE0004: Could not load the framework 'IDEDistribution' (path: /Applications/Xcode.app/Contents/SharedFrameworks/IDEDistribution.framework/Versions/A/IDEDistribution): 
```

### Version with bug

8.0.82 SR8.2

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

None

### Affected platforms

iOS

**Comment by BrandanN21** [comment]
I was able to downgrade to Xcode 15.4 and continue developing with Net 8. 

Our team was able to find this documentation on the Apple Developer website stating that the requirements for iOS App submissions is at a minimum developing with Xcode 15. Starting April 2025 it would be required to use the iOS 18 SDK (xcode 16).

Source 1: https://developer.apple.com/ios/submit/ 
Source 2: https://developer.apple.com/news/?id=utw4yhtp

I also turned off automatic updates for the app store so xcode won't update overnight with us knowing. 

Hope this helps!

**Comment by BrandanN21** [comment]
@lszl84 In case you need it you can install xcode 15.4 from this link: https://developer.apple.com/download/all/

Once it is installed you need to ensure that your command line tools point to 15.4 instead of 16. I am able to have both xcode 16 and 15.4 installed on my machine at the same time. 

That should get you back up and running!

**Comment by tj-devel709** [comment]
Thank you for the feedback @lszl84. As @BrandanN21 pointed out, there is a issue on the xamarin-macios repo tracking this issue already. Please refer to that thread there for further updates.

**Comment by Kremed** [comment]
> @lszl84 In case you need it you can install xcode 15.4 from this link: https://developer.apple.com/download/all/
> 
> Once it is installed you need to ensure that your command line tools point to 15.4 instead of 16. I am able to have both xcode 16 and 15.4 installed on my machine at the same time.
> 
> If you are on MacOS Sequoia already someone has posted a workaround for issues with that. [xamarin/xamarin-macios#20802 (comment)](https://github.com/xamarin/xamarin-macios/issues/20802#issuecomment-2356393897)

My app has been rejected five times by the Apple App Review team, with a lot of unclear logs that seem to indicate crashes caused by some UI library or something similar. I released my app using Xcode 15.4. Could this be the reason for these rejections?

**Comment by softlion** [comment]
> I released my app using Xcode 15.4. Could this be the reason for these rejections?

no
